pkg/ccl/changefeedccl/enriched_source_provider.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package changefeedccl
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/avro"
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdcevent"
+	"github.com/cockroachdb/cockroach/pkg/util/json"
+)
+
+type enrichedEnvelopeSourceProviderOpts struct {
+	updated, mvccTimestamp bool
+}
+type enrichedEnvelopeSourceDataSource struct {
+	jobId string
+}
+type enrichedEnvelopeSourceProvider struct {
+	opts       enrichedEnvelopeSourceProviderOpts
+	sourceData enrichedEnvelopeSourceDataSource
+	// TODO(#139692): Add schema info support.
+	// TODO(#139691): Add job info support.
+	// TODO(#139690): Add node/cluster info support.
```

Comment:
probably these comments should be in the `enrichedEnvelopeSourceDataSource` struct